relay.go

```diff
@@ -654,6 +667,63 @@ func (r *Relayer) handleLocalCallReq(cr *lazyCallReq) bool {
 	return true
 }
 
+func (r *Relayer) fragmentingSend(call RelayCall, f *lazyCallReq, relayToDest relayItem, origID uint32) error {
+	// TODO(echung): should we pool the writers?
+	fragWriter := newFragmentingWriter(
+		r.logger, r.newFragmentSender(relayToDest.destination, f, origID, call),
+		f.checksumType.New(),
+	)
+	defer r.conn.opts.FramePool.Release(f.Frame)
+
+	if len(f.arg2Appends) > 0 && f.isArg2Fragmented {
+		return errFragmentedArg2WithAppend
```

Comment:
`len(f.arg2Appends) > 0` is assumed if we're in this method right?

Do we have any unit tests for this case?

Reply:
the assumption was that we'd be reusing this for `Arg2Set()` in the future as well. the name of the method also doesn't explicitly say `append` in it, so I think being explicit about what this check is for is better